### PR TITLE
Splines small classes doc

### PR DESCRIPTION
**Pull request by blegouix**

### CODE REVIEW

File: include/ddc/kernels/splines/spline_boundary_conditions.hpp [hunk 1]
```diff
@@ -8,16 +8,22 @@
...
-    // Periodic boundary condition u(1)=u(n)
-    PERIODIC,
-    // Hermite boundary condition
-    HERMITE,
-    // Use Greville points instead of conditions on derivative for B-Spline
-    // interpolation
-    GREVILLE,
+    PERIODIC, ///< Periodic boundary condition u(1)=u(n)
+    HERMITE, ///< Hermite boundary condition
+    GREVILLE, ///< Use Greville points instead of conditions on derivative for B-Spline interpolation
```
Comment (tpadioleau) [inline include/ddc/kernels/splines/spline_boundary_conditions.hpp:16]: I think for these two boundary conditions we will need at least a reference to explain what they are ?

File: include/ddc/kernels/splines/spline_boundary_conditions.hpp [hunk 2]
```diff
@@ -32,6 +38,14 @@ static inline std::ostream& operator<<(std::ostream& out, ddc::BoundCond const b
...
 
+/**
+ * @brief Return the number of equations needed to describe a given boundary condition.
+ *
+ * @param bc The boundary condition.
+ * @param degree The degree of the spline.
+ *
+ * @return The number of equations.
+ **/
 constexpr int n_boundary_equations(ddc::BoundCond const bc, std::size_t const degree)
```
Comment (tpadioleau) [inline include/ddc/kernels/splines/spline_boundary_conditions.hpp:49]: I believe `constexpr` and exceptions are not compatible.

**Review by tpadioleau — APPROVED** [review]
Good to me

File: include/ddc/kernels/splines/spline_boundary_conditions.hpp (hunk 1 above)
Comment (blegouix) [reply include/ddc/kernels/splines/spline_boundary_conditions.hpp:16]: In #437 there is a link to @EmilyBourne thesis. Maybe we can write it there too ? Or Emily has another proposal ?

Comment (tpadioleau) [reply include/ddc/kernels/splines/spline_boundary_conditions.hpp:16]: Yes seems good to me

**Review by tpadioleau — APPROVED** [review]

**Pull request merged**